src/ir/comp.rs

```diff
@@ -9,6 +9,7 @@ use crate::utils::Idx;
 #[derive(Default)]
 pub struct Context {
     pub comps: IndexStore<Component>,
+    pub entrypoint: Option<CompIdx>,
```

Comment:
This shouldn't be an optional by the time we hit the IR. Instead, if the user did not provide an entry point, we should look for "main" and if it missing, then we should throw an error

Reply:
I left the Option in case in the future we wanted to have the ability to compile without entrypoint, but I can remove that for now.

Reply:
Further issue: `Default` cannot be derived if the option is not included, as the entrypoint becomes required. Should we remove this trait altogether and just add a new function?